### ISSUE TITLE
Microsoft.SerialConsole/serialPorts/read operation missing

### ISSUE THREAD

**Issue by tyson-trust** [Azure/azure-rest-api-specs#23472]
In [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/serialconsole/resource-manager/Microsoft.SerialConsole/stable/2018-05-01/serialport.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/serialconsole/resource-manager/Microsoft.SerialConsole/stable/2018-05-01/serialport.json) for the operation `SerialPorts_Get` you need the permission `Microsoft.SerialConsole/serialPorts/read` however this operation does not exist. If I try the API without a permission of `Microsoft.SerialConsole/serialPorts/*` I get the error message:
```
{
    "error": {
        "code": "AuthorizationFailed",
        "message": "The client 'redacted' with object id 'redacted' does not have authorization to perform action 'Microsoft.SerialConsole/serialPorts/read' over scope '/subscriptions/<redacted>/resourcegroups/compute/providers/Microsoft.Compute/virtualMachines/lin/providers/Microsoft.SerialConsole/serialPorts/0' or the scope is invalid. If access was recently granted, please refresh your credentials."
    }
}
```
Yet List Operations as documented in serialconsole.json does not return Microsoft.SerialConsole/serialPorts/read as an operation

**Comment by SwathiDhanwada-MSFT** [comment]
@tyson-trust "Microsoft.SerialConsole/serialPorts/read" isn't an operation , its a permission similar to "Microsoft.SerialConsole/serialPorts/*". By using the wildcard character "*", you are granting the specified permissions to all serial ports within the "Microsoft.SerialConsole" resource provider.

For example, if you specify "Microsoft.SerialConsole/serialPorts/read" in a role definition, you are granting the specified permissions to read data from all serial ports within the "Microsoft.SerialConsole" resource provider. Similarly, if you specify "Microsoft.SerialConsole/serialPorts/write" in a role definition, you are granting the specified permissions to write data to all serial ports within the "Microsoft.SerialConsole" resource provider.

This Microsoft.SerialConsole/serialPorts/* contains below permissions 

- Microsoft.SerialConsole/serialPorts/write
- Microsoft.SerialConsole/serialPorts/read
- Microsoft.SerialConsole/serialPorts/delete

**Comment by SwathiDhanwada-MSFT** [comment]
@tyson-trust Could you kindly confirm if you have had the opportunity to review my comment? If you have any questions, please don't hesitate to ask.

**Comment by SwathiDhanwada-MSFT** [comment]
@tyson-trust Could you kindly confirm if you have had the opportunity to review my comment? If you have any questions, please don't hesitate to ask.

**Comment by tyson-trust** [comment]
Hi @SwathiDhanwada-MSFT, Thanks for following up. The problem is I cannot add Microsoft.SerialConsole/serialPorts/read as a permission and I am forced to add Microsoft.SerialConsole/serialPorts/* instead, which does not meet the principle of least privilege 
![Screenshot 2023-05-02 at 16 13 11](https://user-images.githubusercontent.com/111938200/235592384-115a56bb-2481-4dda-9a83-278e2ef3fcf1.png)

**Comment by SwathiDhanwada-MSFT** [comment]
@tyson-trust Indeed, we should always follow the principle of least privilege. Currently, the permission you are looking for does not exist. However, I am checking internally if this can be added. I will keep you posted. Thanks.

**Comment by tyson-trust** [comment]
Ok thanks please let me know how this progresses

Regards
Tyson

CTO

***@***.***
+61 487 747 574
Need a meeting with me? https://calendly.com/tyson-trust/30min
